### DISCUSSION TITLE
Auth, session, token and cookies

### DISCUSSION THREAD

**Discussion by nicopean** [serverpod/serverpod#2359] [Q&A]
Hello, I started a project using serverpod (2.0.0) and a flutter mobile app, everything worked like a charm till I tried to add to that project a web page (made with Flutter), with an authentification form to restrict or give access through routes. But for now i can't make it working with the session and request infos I get.

From that web page, which is just a form using the auth module, if if sign in, I see auth datas stored localy in Chrome . Right.
And I see also serverpod debug logs saying the user is authenticated. Good.

From printed session and request infos within Serverpod's RouteRoot widget and everytime I refresh the page I get :

session.isSigned is false
session.startime is alway a new date
session.duration obviously near 0

but
request.cookies is there with the ssid, 
request.session.isNew = false

To handle routes the way I want, these datas sounds inconsistent to me. Do I miss something? Or do I need to handle myself auth by a cookie token?

Thanks in advance for reading this and to anyone that could help me.

**Comment by SandPod** [comment]
Hey @nicopean!

For clarification, is the endpoint you are requesting a Relic server endpoint?
If so, for authentication to work, the request needs to contain a cookie with the authentication key stored under the `'auth'` keyword.

https://github.com/serverpod/serverpod/blob/26208b33d60b0b76064bad6c30e1b59d81e4a5b4/packages/serverpod/lib/src/relic/web_server.dart#L122-L127

Relic is still experimental, so the experience there might not be ideal. The plans for improvements are tracked here: https://github.com/serverpod/serverpod/issues/2008

**Comment by nielsenko** [comment]
In serverpod 3.x you can get the `Session` (and via that the `AuthenticationInfo?`) from the request.
```dart
final authInfo = (await req.session).authenticated;
```
